### PR TITLE
don't copy over bzlib.h permissions in build.rs

### PR DESCRIPTION
**Pull request by csm123199**
TL;DR:
I had a failure building bzip2-rs (through the zip crate) on an external unix drive/mount, due to chmod not working on it. This patch uses [`std::io::copy`](https://doc.rust-lang.org/std/io/fn.copy.html) instead of [`std::fs::copy`](https://doc.rust-lang.org/std/fs/fn.copy.html) to prevent copying over old file metadata/permissions/etc, and instead just copies over the file's contents, using the default destination's permissions.

rationale/how I found this:
Specifically, this EPERM error was caused [due to me not being the owner of the drive/mount I'm working on](https://www.man7.org/linux/man-pages/man2/chmod.2.html#ERRORS) (owned by root, but I am within a group that can read/write/exe all the files). This allows me to create/modify/etc all the files themselves, but not change the permissions (also NTFS, so it wouldn't make sense to anyway).

The error isn't really clear from the build.rs error output either (viewing the backtrace didn't help):
```
running: "ar" "s" "/media/ext_drive/my_crate/target/debug/build/bzip2-sys-60b04b335573cd8e/out/lib/libbz2.a"
  exit code: 0
  cargo:rustc-link-lib=static=bz2
  cargo:rustc-link-search=native=/media/ext_drive/my_crate/target/debug/build/bzip2-sys-60b04b335573cd8e/out/lib

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" }', /home/$USER/.cargo/registry/src/github.com-1ecc6299db9ec823/bzip2-sys-0.1.9+1.0.8/build.rs:43:64
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
``` 

I had to strace it, and eventually traced it back to [this line in the stdlib](https://github.com/rust-lang/rust/blob/master/library/std/src/sys/unix/fs.rs#L1200):
```
19144 openat(AT_FDCWD, "/media/ext_drive/my_crate/target/debug/build/bzip2-sys-60b04b335573cd8e/out/include/bzlib.h", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0100664) = 8
19144 statx(8, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_BASIC_STATS, stx_attributes=0, stx_mode=S_IFREG|0770, stx_size=0, ...}) = 0
19144 fchmod(8, 0100664)                = -1 EPERM (Operation not permitted)
19144 close(8)                          = 0
```

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks for the PR! This seems fine to me but it feels sort of weird, is this the only reason something doesn't build on your mount? Not supporting `fchmod` seems like it'd cause a lot of things to not build?

**Comment by csm123199** [comment]
I'm not sure what other crates have this sort of problem (I guess generic to using fs::copy inside build.rs), but I run into this building my [tarsum](https://crates.io/crates/tarsum) crate on an external NTFS drive. Building off my home folder (regular, ext4/etc permissions) works just fine.

As the machine is essentially a NAS, I have it setup so `root` owns all the files, and r/w/x access is controlled via a group, `storage`. Here's the drive's fstab line:
```
UUID=XXXXXXXXXXXXXXXX /media/2tb ntfs-3g   defaults,nls=utf8,umask=0000,dmask=0007,fmask=0007,uid=root,gid=storage,windows_names 0 0
```
This results in all files having permissions like the following:
```
-rwxrwx--- 1 root storage 1121 Dec 14 15:44 Cargo.lock
-rwxrwx--- 1 root storage  307 Dec 14 15:44 Cargo.toml
drwxrwx--- 1 root storage    0 Dec 14 15:43 src
drwxrwx--- 1 root storage  376 Dec 14 15:44 target
```
Since all the files are owned by `root` (but rwx by everyone in the `storage` group), chmod fails. I imagine this can be easily tested with an NTFS ramdisk (or probably something less extreme), if necessary.

Creating a new cargo project with just a dependency of `bzip2 = "0.4.1"` doesn't build due to fchmod, but `bzip2 = { git = "https://github.com/csm123199/bzip2-rs" }` does.

You're right - not supporting chmod breaks more things, like git. I looked more into how I can handle this, and it turns out ntfs-3g should have a `silent` option to hide chmod errors like these (and even `mount` has a `quiet` option for this), but I couldn't get those mount options to work.

Since even git chokes on this shared filesystem setup, I'd understand if this is out of scope or something; I personally just switched drives to fix it for me :)

**Comment by alexcrichton** [comment]
Ok thanks for the info! In that case I think I'd prefer to go ahead and close this because I'd prefer to not have a specific workaround like this in the build script. If this becomes too problematic for you, though, this can also be added.
